### PR TITLE
[#292] During channel initialisation the version of resolved manifest is made available

### LINKED ISSUE #292: Ability to retrieve version of the manifest (if Maven manifest) use by the session
In case manifest coordinates given to ChannelSession are Maven GA coordinates, prospero might be interested in knowing what is the actual resolved version of the manifest(s).

The use case is that during the update operation, prospero would compare current and target versions of the manifest(s), and refused the update operation (or requested further confirmation) if the update constitutes a downgrade to an earlier manifest version.

Cc @spyrkob for your review too.

### PR DESCRIPTION
**Pull request by spyrkob**
Fixes #292

### CODE REVIEW

**Comment by spyrkob** [comment]
@TomasHofman @jmesnil could you review this please?

**Comment by TomasHofman** [comment]
Great, will check tomorrow!

**Review by TomasHofman — APPROVED** [review]
Looks good to me :+1:

File: core/src/main/java/org/wildfly/channel/ChannelImpl.java [hunk 1]
```diff
@@ -95,11 +102,14 @@ void init(MavenVersionsResolver.Factory factory, List<ChannelImpl> channels) {
 
         if (channelDefinition.getBlocklistCoordinate() != null) {
             BlocklistCoordinate blocklistCoordinate = channelDefinition.getBlocklistCoordinate();
+            resolvedChannelBuilder.setBlocklistCoordinate(blocklistCoordinate);
```
Comment (TomasHofman) [inline core/src/main/java/org/wildfly/channel/ChannelImpl.java:105]: Btw here it looks we are just reusing the original blocklist coordinate, without resolving the version..?

(I don't need resolved blocklist, just point it out if it's intentional.)

**Comment by TomasHofman** [comment]
@parsharma this PR would introduce API you need to obtain manifest version from channel session. See the `channelSession.getRuntimeChannels()` method.

File: core/src/main/java/org/wildfly/channel/ChannelImpl.java (hunk 1 above)
Comment (spyrkob) [reply core/src/main/java/org/wildfly/channel/ChannelImpl.java:105]: Thanks, good catch

**Review by jmesnil — APPROVED** [review]

**Pull request merged**